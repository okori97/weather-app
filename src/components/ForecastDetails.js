import React from "react";
import GetDate from "../helpers/getDate";
import addSuffix from "../helpers/addSuffix";
import "../styles/ForecastDetails.css";

export default function ForecastDetails({ forecasts }) {
  const { date, temperature, humidity, wind } = forecasts[0];
  console.log(forecasts[0]);

  // Converts unix into date and adds suffix to month
  const { day, monthNum, monthString } = GetDate(date);
  let convertedDate = `${day} ${addSuffix(monthNum)} ${monthString}`;

  return (
    <div className="forecast-details">
      <h2>{convertedDate}</h2>
      <div className="forecast-details--item">
        <span className="bold">Max Temperature: </span>
        {`${temperature.max}°c`}
      </div>
      <div className="forecast-details--item">
        <span className="bold">Min Temperature: </span>
        {`${temperature.min}°c`}
      </div>
      <div className="forecast-details--item">
        <span className="bold">Humidity: </span>
        {`${humidity}%`}
      </div>
      <div className="forecast-details--item">
        <span className="bold">Wind: </span>
        {wind.speed}
      </div>
    </div>
  );
}
