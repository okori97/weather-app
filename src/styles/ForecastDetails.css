.bold {
  font-weight: bold;
}

.forecast-details {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.forecast-details--item {
  margin-bottom: 8px;
}

.forecast-details--item,
p {
  font-size: 14px;
}
