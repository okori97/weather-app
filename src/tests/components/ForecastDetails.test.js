import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: "111",
    temperature: {
      max: 12,
      min: 4,
    },
    icon: 800,
    wind: {
      speed: 60,
    },
    humidity: 40,
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct value for props", () => {
    const { getByText } = render(<ForecastDetails forecasts={validProps} />);

    expect(getByText("Thu 1st Jan")).toBeInstanceOf(HTMLHeadingElement);
    expect(getByText("12°c")).toHaveClass("forecast-details--item");
    expect(getByText("4°c")).toHaveClass("forecast-details--item");
    expect(getByText("60")).toHaveClass("forecast-details--item");
    expect(getByText("40%")).toHaveClass("forecast-details--item");
  });
});
