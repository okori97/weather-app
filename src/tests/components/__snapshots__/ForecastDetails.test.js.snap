// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h2>
      Thu 1st Jan
    </h2>
    <div
      class="forecast-details--item"
    >
      <span
        class="bold"
      >
        Max Temperature: 
      </span>
      12°c
    </div>
    <div
      class="forecast-details--item"
    >
      <span
        class="bold"
      >
        Min Temperature: 
      </span>
      4°c
    </div>
    <div
      class="forecast-details--item"
    >
      <span
        class="bold"
      >
        Humidity: 
      </span>
      40%
    </div>
    <div
      class="forecast-details--item"
    >
      <span
        class="bold"
      >
        Wind: 
      </span>
      60
    </div>
  </div>
</DocumentFragment>
`;
